crates/blockchain/payload.rs

```diff
@@ -325,10 +330,49 @@ pub fn fill_transactions(context: &mut PayloadBuildContext) -> Result<(), ChainE
 
 /// Executes the transaction, updates gas-related context values & return the receipt
 /// The payload build context should have enough remaining gas to cover the transaction's gas_limit
-// TODO(https://github.com/lambdaclass/ethereum_rust/issues/678): Handle blobs in blob txs
 fn apply_transaction(
     head: &HeadTransaction,
     context: &mut PayloadBuildContext,
+) -> Result<Receipt, ChainError> {
+    match head.tx {
+        Transaction::EIP4844Transaction(_) => apply_blob_transaction(head, context),
+        _ => apply_plain_transaction(head, context),
+    }
+}
+
+/// Runs a blob transaction, updates the gas count & blob data and returns the receipt
+fn apply_blob_transaction(
+    head: &HeadTransaction,
+    context: &mut PayloadBuildContext,
+) -> Result<Receipt, ChainError> {
+    // Fetch blobs bundle
+    let tx_hash = head.tx.compute_hash();
+    let Some(blobs_bundle) = context.store().get_blobs_bundle_from_pool(tx_hash)? else {
+        // No blob tx should enter the mempool without its blobs bundle so this is an internal error
+        return Err(
+            StoreError::Custom(format!("No blobs bundle found for blob tx {tx_hash}")).into(),
+        );
+    };
+    if (context.blobs_bundle.blobs.len() + blobs_bundle.blobs.len()) as u64 * GAS_PER_BLOB
+        > MAX_BLOB_GAS_PER_BLOCK
+    {
+        // This error will only be used for debug tracing
+        return Err(EvmError::Custom("max data blobs reached".to_string()).into());
+    };
+    // Apply transaction
+    let receipt = apply_plain_transaction(head, context)?;
+    // Update context with blob data
+    let prev_blob_gas = context.payload.header.blob_gas_used.unwrap_or_default();
```

Comment:
Is it correct to default here? That would be zero, right?

Reply:
Yes, if the blob gas was not set then it means no blob transactions have been ran and no blob gas has been used